src/wp-includes/rest-api/endpoints/class-wp-rest-font-faces-controller.php

```diff
@@ -0,0 +1,941 @@
+	 * @since 6.5.0
+	 *
+	 * @param int $font_family_id Supplied ID.
+	 * @return WP_Post|WP_Error Post object if ID is valid, WP_Error otherwise.
+	 */
+	protected function get_parent_font_family_post( $font_family_id ) {
+		$error = new WP_Error(
+			'rest_post_invalid_parent',
+			__( 'Invalid post parent ID.', 'default' ),
+			array( 'status' => 404 )
+		);
+
+		if ( (int) $font_family_id <= 0 ) {
+			return $error;
+		}
+
+		$font_family_post = get_post( (int) $font_family_id );
+
+		if ( empty( $font_family_post ) || empty( $font_family_post->ID )
+		|| 'wp_font_family' !== $font_family_post->post_type
+		) {
+			return $error;
+		}
+
+		return $font_family_post;
+	}
+
+	/**
+	 * Prepares links for the request.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param WP_Post $post Post object.
+	 * @return array Links for the given post.
+	 */
+	protected function prepare_links( $post ) {
+		// Entity meta.
+		return array(
+			'self'       => array(
+				'href' => rest_url( $this->namespace . '/font-families/' . $post->post_parent . '/font-faces/' . $post->ID ),
+			),
+			'collection' => array(
+				'href' => rest_url( $this->namespace . '/font-families/' . $post->post_parent . '/font-faces' ),
+			),
+			'parent'     => array(
+				'href' => rest_url( $this->namespace . '/font-families/' . $post->post_parent ),
+			),
+		);
+	}
+
+	/**
+	 * Prepares a single font face post for creation.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 * @return stdClass|WP_Error Post object or WP_Error.
+	 */
+	protected function prepare_item_for_database( $request ) {
+		$prepared_post = new stdClass();
+
+		// Settings have already been decoded by ::sanitize_font_face_settings().
+		$settings = $request->get_param( 'font_face_settings' );
+
+		// Store this "slug" as the post_title rather than post_name, since it uses the fontFamily setting,
+		// which may contain multibyte characters.
+		$title = WP_Font_Utils::get_font_face_slug( $settings );
+
+		$prepared_post->post_type    = $this->post_type;
+		$prepared_post->post_parent  = $request['font_family_id'];
+		$prepared_post->post_status  = 'publish';
+		$prepared_post->post_title   = $title;
+		$prepared_post->post_name    = sanitize_title( $title );
+		$prepared_post->post_content = wp_json_encode( $settings );
+
+		return $prepared_post;
+	}
+
+	/**
+	 * Sanitizes a single src value for a font face.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $value Font face src that is a URL or the key for a $_FILES array item.
+	 *
+	 * @return string Sanitized value.
+	 */
+	protected function sanitize_src( $value ) {
+		$value = ltrim( $value );
+		return false === wp_http_validate_url( $value ) ? (string) $value : sanitize_url( $value );
+	}
+
+	/**
+	 * Handles the upload of a font file using wp_handle_upload().
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param array $file Single file item from $_FILES.
+	 * @return array Array containing uploaded file attributes on success, or error on failure.
```

Comment:
Since this method can also return a `WP_Error`, this needs to be adjusted accordingly.

```suggestion
	 * @return array|WP_Error Array containing uploaded file attributes on success, or error on failure.
```

Reply:
Updated in https://github.com/WordPress/wordpress-develop/pull/6034/commits/a0836f0f6b4fffc3259a2711f2ea43b02823a6e2